pkg/build/nodeimage/imageimporter.go

```diff
@@ -53,9 +54,16 @@ func (c *containerdImporter) Pull(image, platform string) error {
 }
 
 func (c *containerdImporter) LoadCommand() exec.Cmd {
-	return c.containerCmder.Command(
+	var opt string
+	// TODO: Hack to workaround #3795.
+	if runtime.GOARCH == "amd64" {
+		opt = "--local=false"
```

Comment:
can you elaborate a bit on what this is intended to do and why only amd64?

it will be a bit before I can dig into this

Reply:
I dug a bit more into the actual issue on the issue thread, which will hopefully answer how to actually fix this in containerd / docker (which I do believe it is actually a bug in).

The reason for the arm64 / amd64 split is confusing and what makes this a hack:

* This is happening with packages where there is an amd64 / 386 mix, hence why targeting amd64 for `--local=false`.
* On the version of containerd kind is currently using `--all-platforms` with `--local=false` doesn't work.

So the bug has been seen on amd64, people using arm64 do want to mix platforms. It's also possible for arm64 to mix v8 or v9, but I think that's far less common and unlikely to break anyone.

Clearly this should be fixed in containerd (or docker, if it should actually include more blobs in its images), but given a kind release is tied to a containerd release I think it would be safe to include this hack until kind is on a containerd with a fix.

Reply:
I don't like much the idea of we carry patching compensation for known bugs

Reply:
As a general rule we actually permit using them across releases and notify when a release has a compatibility change that requires images built after a certain release or something like that. It hasn't happened recently as well.